test/agenda.js

```diff
@@ -503,4 +505,50 @@ describe('Agenda', () => {
       expect(jobs3).to.have.length(1);
     });
   });
+
+  describe('transactions', () => {
+    before(async function () {
+      if(!mongoURI.includes('replicaSet')){
```

Comment:
I guess the condition should relies on v4 checking

Reply:
How to check if that's v4?

Reply:
Not sure in units tests, but I mongo generally you can use this: https://stackoverflow.com/a/16932830/5320409

Maybe we can add a check at instantiation that turn a value in the root class and throw an error if you try to pass a `session` object on <v4 function ?

Tell me if you've a better idea ? :)

Reply:
Your suggestions are perfect.